plugin/pkg/scheduler/scheduler.go

```diff
@@ -44,6 +50,33 @@ type Scheduler struct {
 	config *Config
 }
 
+// These are the functions which need to be provided in order to build a Scheduler configuration.
+// An implementation of this can be seen in factory.go.
+type Configurator interface {
```

Comment:
ha "go java" well.. its a `configurator` because it helps to configure with actually being a configuration data object.